### PR TITLE
feat(ui): Make file types themeable

### PR DESCRIPTION
**Pull request by cfxegbert**
Addresses issue https://github.com/eza-community/eza/issues/235.

### CODE REVIEW

**Comment by cfxegbert** [comment]
Just noticed https://github.com/eza-community/eza/pull/63/files.  Should I refactor to be more like #63? Should I rename immediate to build?

**Comment by cafkafk** [comment]
> Should I refactor to be more like https://github.com/eza-community/eza/pull/63?

If you wanted to, perhaps you could do so in a follow up PR (which would make it easier to see the diff). I don't think you have too, but maybe I'm missing something.

> Should I rename immediate to build?

Build is probably more intuitive, immediate might be more "correct". I may lean slightly to being more intuitive, but I think you should make the pick you prefer tbh.

**Comment by PThorpe92** [comment]
> If you wanted to, perhaps you could do so in a follow up PR (which would make it easier to see the diff).

I have been trying to diff your PR and the old one by switching tabs. and I would heavily agree with this.

I would say that if there is anything you feel 63 does better / cleaner than yours, implement it and they can be essentially merged into one, and we can close 63 in favor of this one :+1:

File: src/theme/ui_styles.rs [hunk 1]
```diff
@@ -121,6 +122,20 @@ pub struct SecurityContext {
     pub selinux: SELinuxContext,
 }
 
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct FileType {
```
Comment (cafkafk) [inline src/theme/ui_styles.rs:126]: I'd prefer documentation on this

File: src/theme/mod.rs [hunk 2]
```diff
@@ -144,14 +143,14 @@ impl Definitions {
 }
 
 
-pub trait FileColours: std::marker::Sync {
-    fn colour_file(&self, file: &File<'_>) -> Option<Style>;
+pub trait FileColours: Sync {
```
Comment (cafkafk) [inline src/theme/mod.rs:146]: Would you consider documenting this trait and `colour_file`? Just for posterity

**Review by cafkafk — APPROVED** [review]
LGTM :+1:, haven't tested this but have tested that it doesn't seem to break anything.

Sidequest: While you have these parts of the code fresh in your mind, would you like to document them? If not just ping me and I'll merge.

**Review by cafkafk — APPROVED** [review]
Still works for me, and integration tests work, so merging, ty for the PR

**Pull request merged**